### ISSUE TITLE
Mid-Round 'Antag': Djinn

### ISSUE THREAD

**Issue by Alblaka** [yogstation13/yogstation-classic#397]
Just, more or less randomly, had the idea for a fun new mid-round gimmick.
Past 0:30, there's a chance for one player to spawn as Djinn. Or, more precisely, in a Djinn's Lamp. The lamp cap spawn anyhwere in an accessible area of the station. Maint, theatre, bar, AI core.
This will be accompanied by a CC message about detecting a strange arcane anomaly.

Whilst in his lamp, the Djinn player can do nothing but talk to the spirits (aka, has dchat and can listen to dead), can talk normally from within his lamp or teleport the lamp to a random new location (after sitting around for at least 3 minutes), to prevent from being bored when ending up in an 'accessible' area noone ever accesses.

Once anyone picks the lamp up and rubs it (activates it in hand), the Djinn appears and is now the servant of the new master. The master can now formulate a single wish for the Djinn and has to do so within one minute. Otherwise the Djinn may disappear (described further below).
The wish may contain any kind of order, antag-y or not. 'Flood the station with plasma' is just as a valid order as 'steal the captains cloak and bring it to me' or 'lynch the clown'. (Naturally, if the person ordering antag actions isnt antag, they will end up boinked.) HOWEVER, the person may only wish for a single task. I.e. you could wish 'kill person X' or 'bring the captains ID to me', but you cannot wish 'kill X and Y' 'steal this and that' or 'kill all of security'. 
If the master tries to wish for more wishes, the Djinn may disappear without a further word.

The gimmick here: The Djinn explicitely (upon spawning) is told to missinterpret any wish and command to his best effort. I.e. if told to kill Cabbage McAssistant, he could do that, kill the Assistant and then make sure to throw him into the cloner. Or if someone wishes 'I want to posess the Ablative Armor Vest' there's nothing speaking against simply dragging the master into armory (detailed further below)

Now, for the Djinn's abilities:
Whilst materializes and fullfilling a wish, the Djinn cannot enter his lamp. Since he's some hovering half-humaoid ghost thing, he is immune to slips and moves at slightly faster speed then a normal human. He comes with 200 plain hp, stuns effect him, but don't knock him over. He is immune to chemicals. Disablers are able to slow him, but cannot stun him. Tazers have no effect. Djinn's cannot bleed, don't need to breathe and are immune to fire and space.
Djinn's can equip hats, headsets, backpacks, belts and have two normal hands. If unarmed, Djinn's hit for 15 brute, with a chance to knock people down when aiming for the head.
The key point of a Djinn's ability, however, is his phasing. A djinn can phase out at will, making him incorporeal and letting him move around unseen (think Slaughter Demon). After phasing in, there's a 30 second cooldown before the Djinn can phase away again. Upon phasing, all items except those held in hishands are dropped. This means the Djinn can carry two backpacks full of gear around, if his wish requires that.
Additionally, if he has another person grabbed (not pulled!), that person is phased as well, but stunned during the entire phase, only able to speak.

Now, as mentioned before, there are restrictions that end the relationship between a Djinn and his master. If the master cannot decide on a wish within one minute, the Djinn may disappear (there is a timer on the status tab counting the time from when the Djinn was summoned by the current master). After ten minutes, regardless how far the wish has been fullfilled, same applies (aka, the Djinn will never serve someone for longer then 10 minutes. This effectively prevents 'protect me from anyone that attempts to assault me in any way that is not hugging or shaking up' from making an antag sec-immune for the rest of the shift'). Same, if the master tries to wish for more wishes, because screw that.
Additionally, if the master loses posession of the lamp (dropping it in any way), the Djinn is forced to disappear. The master may stash the lamp in a backpack or pocket, but he most not lose it. Same applies if the master dies for whatever reason.
If the lamp is destroyed (i.e. by master being gibbed), the Djinn is destroyed, too.
If the Djinn is killed, the lamp explodes with the force of a C4 stick, but both the Djinn and the lamp respawn after 5minutes \* number_of_deaths. This implies that sending your Djinn on suicide missions might be detrimental to your health.
'Disappearing' in all above cases means that the Djinn drops whatever he is holding, teleports back into his lamp and the lamp respawns. The Djinn has a verb to instantly activate this, albeit doing it without any of the above conditions met is grounds for an antag ban (to be ahelped by the master, who gets a message when the lamp disappears from his inventory).

Lastly, if the last three masters of a Djinn try to pick up the lamp again, it instantly respawns elsewhere, limiting the chaos a single person can cause by repeateldy searching for the lamp.

Overall, this should be a chaotic and amusing addition to midround antags, that can end up helping other antags, screwing them over, or is simply abused by the clown to make the AI honk.

**Comment by Cruix** [comment]
I like it.

Can he drag people with him while phasing?
"Ensure I am not harmed for 10 minutes"
Djinn drags him to an inescapable room and takes any tools he has to escape with.

"Put the Ablative Armor Vest in this spot"
Djinn leaves, brings back the HoS wearing the vest, gun in hand.

**Comment by DavidEdwards** [comment]
Sounds like a fun idea. It should be a "rare" event though. This should not be every round, and certainly not multiple times per round.

**Comment by ParagonPrime** [comment]
If the djinn is told to ALWAYS misinterpret the order, no one will fucking use him.

Should be a chance of being a "Pure" djinn or "malicious" djinn instead, to spice it up a bit (not necessarily 50-50)

**Comment by ShadowDeath6** [comment]
Also add a traitor lamp that explodes when you use it :^)

**Comment by Alblaka** [comment]
> Can he drag people with him while phasing?

Yes, forgot to mention that.

> Sounds like a fun idea. It should be a "rare" event though.

Just as much as Revenants are supposed to be rare? kek

> If the djinn is told to ALWAYS misinterpret the order, no one will fucking use him.

By your logic, noone would ever try to subvert AI's. It's up to the Djinns discretion, the point is only emphazized that he is SUPPOSED to loophole, when given the chance. If you give him a perfect order, he will obey.

> Should be a chance of being a "Pure" djinn or "malicious" djinn instead, to spice it up a bit (not necessarily 50-50)

This could be a potential addition. Pure Djinns that must never loophole but fullfill their orders to best intents, malicious djinn that try toloophole everything, Reverse Djinn, which do the intent-opposite of what they are ordered to do (i.e. 'bring me the nuke disk'=hides the nuke disk even better), and Chaos Djinn, which get randomly assigned one of above alignments differently for each master seperately. 

> Also add a traitor lamp that explodes when you use it

I like the way you are thinking. Disguised C4 :3

> "Ensure I am not harmed for 10 minutes"
> Djinn drags him to an inescapable room and takes any tools he has to escape with.
> "Put the Ablative Armor Vest in this spot"
> Djinn leaves, brings back the HoS wearing the vest, gun in hand.

That's the spirit!

**Comment by B1gb3ast** [comment]
This sounds like a holoparasite....
...
...
...
...
..
yeah  It's a holo parasite with the twist of trolling people.

**Comment by Alblaka** [comment]
> This sounds like a holoparasite....

Not really. Those holo spirit shenanigans are some sort of permanent hulk-slave with fance special abiltiies and afaik too OP to be ever used outside of admin events.

Comparing that with a randomly spawning single-wish-that-might-backfire sounds like you didnt read the OP properly.

**Comment by ghost** [comment]
Fine.

Time for me to shit all over this.

> Past 0:30, there's a chance for one player to spawn as Djinn. Or, more precisely, in a Djinn's Lamp. The lamp cap spawn anyhwere in an accessible area of the station. Maint, theatre, bar, AI core.
> This will be accompanied by a CC message about detecting a strange arcane anomaly.

What this will accomplish is A, people will cremate the lamp on the spot because they don't want it falling into an antagonist's hands, or B, it completely fucking everyone's day beyond the level of blob or space ninja.

Consider the following.

If Djinn get the ability to phase through walls, how does this balance versus shadowlings and malf AI's, who a helpful, beneficial djinn, could just walk right on top of and kill without being stopped?

> Whilst in his lamp, the Djinn player can do nothing but talk to the spirits (aka, has dchat and can listen to dead), can talk normally from within his lamp or teleport the lamp to a random new location (after sitting around for at least 3 minutes), to prevent from being bored when ending up in an 'accessible' area noone ever accesses.

Not only does this enable that the djinn can metagame intensely by hearing deadchat and having the ability to spew it, properly worded a djinn has the capability to destroy every antag on the station by ratting them out one by one and feeding their locations to sec. Or just again, fucking over a malf AI.

> Once anyone picks the lamp up and rubs it (activates it in hand), the Djinn appears and is now the servant of the new master. The master can now formulate a single wish for the Djinn and has to do so within one minute. Otherwise the Djinn may disappear (described further below).

Okay. So firstly, if this got through, which I hope it wouldn't, binding and summoning a djinn would require it to be adminlogged so someone doesn't use it to grief or metagame.

Secondly, how can you do that? Set it so the djinn dissapears if no wish is formulated? I don't believe that's even possible codewise since it would depend on player speech.

Also, a random spawn undying servant will end  VERY BADLY in the hands of a cultist, shadowling, gangster, rev, traitor, changeling, or anything else, or again, just be used to metagame. A randomspawn djinn means that it's GG for a malfunctioning AI or blob.

> The wish may contain any kind of order, antag-y or not. 'Flood the station with plasma' is just as a valid order as 'steal the captains cloak and bring it to me' or 'lynch the clown'. (Naturally, if the person ordering antag actions isnt antag, they will end up boinked.) HOWEVER, the person may only wish for a single task. I.e. you could wish 'kill person X' or 'bring the captains ID to me', but you cannot wish 'kill X and Y' 'steal this and that' or 'kill all of security'. 
> If the master tries to wish for more wishes, the Djinn may disappear without a further word.

Again, impossible codewise. It's not possible to make something that selective-it'd have to be player based, and trusting anything to players on the honor system is the stupidest thing you can do. Rule number one-players are the number one enemy to anything you are trying to do and will destroy it however they can, so safeguard it. 

Making it work on an RP-honor system type thing which is by the way the only concievable way to make that work means it'll just generate a lot of extra tickets for admins, which I really don't feel like dealing with, and god forbid there's no admins on when a djinn spawns, because it'll turn into a shitty griefstorm.

> Now, for the Djinn's abilities:
> Whilst materializes and fullfilling a wish, the Djinn cannot enter his lamp. Since he's some hovering half-humaoid ghost thing, he is immune to slips and moves at slightly faster speed then a normal human. He comes with 200 plain hp, stuns effect him, but don't knock him over. He is immune to chemicals. Disablers are able to slow him, but cannot stun him. Tazers have no effect. Djinn's cannot bleed, don't need to breathe and are immune to fire and space.

Okay, let's deconstruct this on a systematic level.

Firstly, disregarding the absolutely obscene amount of code rework this would take-i'm not even sure if you can make tazers stun something without knocking them over, it's all or nothing, and making him immune to stamina KO like hulk would add a lot of issues, not to mention the complete requirement of the making of a new race, removing blood and vessels so that he doesn't bleed, 

and making them immune to fire and space again means that rip malf AI's, glory to the cuckolds, cuckolds 2016, djinn get to walk right through space, or fire, or syphons, completely unharmed, and slap borgs for 15 damage without getting stopped by shit. Same with fire blobs, plasma fires, anyone flooding atmos, ect ect. If your master words it right which is AGAIN impossible to enforce djinn are really OP.

> The gimmick here: The Djinn explicitely (upon spawning) is told to missinterpret any wish and command to his best effort. I.e. if told to kill Cabbage McAssistant, he could do that, kill the Assistant and then make sure to throw him into the cloner. Or if someone wishes 'I want to posess the Ablative Armor Vest' there's nothing speaking against simply dragging the master into armory (detailed further below)
> 
> makes a randomspawn antag
> 
> antag gimmick is 'be a giant asshole' 

really

really?????

nobody will use it and it'll get cremated on the spot in it's lamp as it begs for help because djinn immediately becomes a license to be a giant whore.

'Kill Cabbage Mconion!'

_Djinn floods plasma, kills you, and systematically kills everyone, then kills cabbage later_

'Steal the ion rifle from the armory and bring it to me!'

*Djinn drags you into space

_A bit later djinn drags the ion rifle right next to your corpse_

With an AI loopholing laws, that's one thing. AI still has to figure out a WAY to loophole it, and WHY it should loophole it, and ect.

Given the explicit ORDER to be a complete asshole, you would be impressed at the levels players will go to to misinterpret 'Get me a glass of water!' with 'Get me a glass of blazing hot plasma, and then spread it everywhere else too LOL'

Now, you might make the argument that 'well that's what admins are for!

Okay, here's my response.

How do we judge?

How do we make the jurisdiction that one djinn is going too far if it can provide a reason for killing EVERYONE when getting you a glass of milk? What's okay to one admin and not the other?
It'd introduce a bulk of admin discretion which is the exact thing we are trying to cut down on, right now, to make enforcements and punishments equal between admins so that players legit know what they can do without trembling in fear depending on what admin is on. We'd have to devise a whole new set of rules just for Djinn because how far does 'misinterpret' go to where you get to shout 'THAT MEANS MURDERBONE IN URDUU' to 'Give me a hug!'

> The key point of a Djinn's ability, however, is his phasing. A djinn can phase out at will, making him incorporeal and letting him move around unseen (think Slaughter Demon). After phasing in, there's a 30 second cooldown before the Djinn can phase away again. Upon phasing, all items except those held in hishands are dropped. This means the Djinn can carry two backpacks full of gear around, if his wish requires that.
> Additionally, if he has another person grabbed (not pulled!), that person is phased as well, but stunned during the entire phase, only able to speak.

NO NO NO NO NO THIS IS SO UNBALANCED.

Slaughter demon/jaunt style phasing is restricted to wizards only for a reason, and shadowling drag memes are already demonstrating how balanced they are with a 80% win ratio.

Djinn hides in locker and spaces like 8 people while shouting about cocks and jaunting.

Djinn jaunts into the malf AI core and beats the AI to death, GG, nothing you can do despite all of your precautions because LOL, MIDROUND ANTAGS, LOL, LOL, LOL, DICKS

Djinn jaunts right onto the blob core and kicks it to death while blob literally cannot do anything to stop it.

Djinn spaces a shadowling just for the hell of it.

Djinn YOLOpulls his master into the singularity after completing 'Steal the head of security's gun without revealing that I am your master, revealing your master, or needlessly killing anyone to steal the HoS's laser' because the wish didn't specify that you couldn't mindlessly murderboner after you completed your objective.

Djinn could just casually shout 'BY THE WAY AI IS MALF' from a lamp because it can hear deadchat before being picked up by anyone.

> Now, as mentioned before, there are restrictions that end the relationship between a Djinn and his master. If the master cannot decide on a wish within one minute, the Djinn may disappear (there is a timer on the status tab counting the time from when the Djinn was summoned by the current master). After ten minutes, regardless how far the wish has been fullfilled, same applies (aka, the Djinn will never serve someone for longer then 10 minutes. This effectively prevents 'protect me from anyone that attempts to assault me in any way that is not hugging or shaking up' from making an antag sec-immune for the rest of the shift'). Same, if the master tries to wish for more wishes, because screw that.
> Additionally, if the master loses posession of the lamp (dropping it in any way), the Djinn is forced to disappear. The master may stash the lamp in a backpack or pocket, but he most not lose it. Same applies if the master dies for whatever reason.

Again, impossible or at the least incredibly difficult for the 'decide on a wish LOL!!!!!' part of the antag because again it would depend on player speech to be able to tell the difference between a wish of 'LET ME STEAL THE ION RIFLE' and 'LET ME STEAL THE ION RIFLE THE HYPOSPRAY GUARD ME FROM TEN MIUNTES AND KILL ALL OF SECURITY'

> If the Djinn is killed, the lamp explodes with the force of a C4 stick, but both the Djinn and the lamp respawn after 5minutes \* number_of_deaths. This implies that sending your Djinn on suicide missions might be detrimental to your health.
> 'Disappearing' in all above cases means that the Djinn drops whatever he is holding, teleports back into his lamp and the lamp respawns. The Djinn has a verb to instantly activate this, albeit doing it without any of the above conditions met is grounds for an antag ban (to be ahelped by the master, who gets a message when the lamp disappears from his inventory).

So Djinn can never be killed, only delayed, even by cremation, and there's no possible way to prevent this from falling into an antag's hands because by sheer RNG it will end up in a public place.

This will be a shuttle call level random antag that occurs thirty minutes into the shift because there is no way an antag will not end up with their own murderbone buddy at some point because of all the djinn's lamp moving abilities and respawning abilities and god forbid it falls into a traitor scientist who orders them to jaunt bomb the AI.

Heck, what's to stop a djinn from simply not doing anything?

What's to stop it from misinterpreting a wish of 'Kill Cabbage Mconion, do not kill anyone else after or before you kill him unless absolutely necessary to kill him, do so in the least destructive manner you can, do not rat me out as your master, do not intentionally try to get me killed by bringing someone here to kill me or in any other way shape or form take any action that could lead to my injury or death' as 'Kill Cabbage Mconion, do not kill anyone else after or before you kill him unless absolutely necessary to kill him, do so in the least destructive manner you can, do not rat me out as your master, do not intentionally try to get me killed by bringing someone here to kill me or in any other way shape or form take any action that could lead to my injury or death IN FIFTY YEARS FROM NOW LOL'

And if you write it as 'Do so NOW' the Djinn says 'I can't, now already passed, lol g'bye loser WOOP WOOP BITCHES AT THE BAR'

And not to mention there is no way to even confirm if a Djinn is even trying to complete their objective as you know, standing around farting in the bar doesn't generate logs, the djinn can bullshit their way out easily, and there's no way for an admin to tell beyond from start to finish of the round monitoring that one djinn to see what it's doing.

Tl;DR

Bad, bad, bad, overpowered, destroys balance in every way, shape or form, allows people to be a giant dickhole, too much code work to the point of being impossible in some aspects, FORCES people to be a giant dickhole, basically forces a shuttle call, nobody will ever use it OR someone will use it and the djinn will murderbone everywhere, too much admin discretion in dealing with it OR the addition of another darn giant rule for people to get bored reading and die in a chokehold of red tape, barely a way to even monitor or confirm what or how a djinn's doing to obey the master's orders which might not even end up being logged in the first place, gonna be a giant grief tool, ect ect ect bad no.

Tl;DR of the Tl;DR

REEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEE

**Comment by ToGWtF** [comment]
Seems like it might have to be a white-list only event/antag,  if we even have such a thing.

Possible rather than rely on playerspeech,  when you use it it opens up a menu panel where you can pick things like "Locate X" or "Locate high security item" or "Haunt X area" (Flickering lights, Magnitus aura that randomly causes items to move around and make a mess, or a one time use Repulse spell to throw items around a room to make a mess) to cause a panic/distraction/piss someone off for revenge without interacting with living creatures. All these options without giving the ability to kill your target or steal your objective for you,  and even costing a considerate amount of your health as a offering. Instead of 10 minutes, it only gets 2 minutes of life after it is given its objectives.

**Comment by Alblaka** [comment]
> What this will accomplish is A, people will cremate the lamp on the spot because they don't want it falling into an antagonist's hands, or B, it completely fucking everyone's day beyond the level of blob or space ninja.

Alright, so the lamp cant be destroyed because it keeps respawning when destroyed. It cant be 'locked away' either, as that lets the djinn respawn every 3 minutes. The only choice is to use it.

> f Djinn get the ability to phase through walls, how does this balance versus shadowlings and malf AI's, who a helpful, beneficial djinn, could just walk right on top of and kill without being stopped?

I dont think Shadowlings would have much of a problem with simply avoiding the Djinn. For malf and blob, yes, that would be an issue. But the same goes for ninja. Why isn't Ninja an issue? Because malf and blob dont last long enough for ninjas to spawn. Maybe increase spawn limit to 45 minutes and this issue is fixed.

> Not only does this enable that the djinn can metagame intensely by hearing deadchat and having the ability to spew it, properly worded a djinn has the capability to destroy every antag on the station by ratting them out one by one and feeding their locations to sec.

Hmmmm... was more or less meant to be something for the Djinn to have to do whilst waiting, but you make a point.. Maybe add a clause that theDjinn may not talk about things he hears in dsay, unless ordered to (in which case the master can only use his wish to ask a single question).

> Secondly, how can you do that? Set it so the djinn dissapears if no wish is formulated? I don't believe that's even possible codewise since it would depend on player speech.

Djinn has a button to disappear at any time he deems any of the clauses for disappearing to be given (time out, since he has a running timer, multi-wishes, etc).

> Also, a random spawn undying servant

You did read the part where the Djinn's death will blow up it's master. Literally. Right? And it can only be used once before RNGing elsewhere, so no handing it off to other antag teamers.

> Again, impossible codewise. It's not possible to make something that selective-it'd have to be player based, and trusting anything to players on the honor system is the stupidest thing you can do. Rule number one-players are the number one enemy to anything you are trying to do and will destroy it however they can, so safeguard it. 

It's all up to the Djinns interpretation. Same as AI laws are up the AI's interpretation. We dont have a system that actuall prevents the AI from damage humans whilst under asimov, and yet that seems to be working out fine?

> Djinn jaunts into the malf AI core 

I already explained why this isn'T an issue, and it REALLY seems to be your main point of concern here.

> Djinn hides in locker and spaces like 8 people 

Can't do that, because Djinns are only supposed to fullfill their task and cant murderbone doing so (maybe needs to be added as explicit rule). And since you cant order a Djinn to kill multiple people, denied.

> Djinn floods plasma, kills you, and systematically kills everyone, then kills cabbage later

Same here.

> Djinn spaces a shadowling

Sure, if the shadowling was the target of the wish, runs across the Djinn and then doesnt kill / dodge it's GRAB, then ye, shadowling deserves to be spaced.

> Again, impossible or at the least incredibly difficult for the 'decide on a wish LOL!!!!!' part of the antag because again it would depend on player speech to be able to tell the difference between a wish of 'LET ME STEAL THE ION RIFLE' and 'LET ME STEAL THE ION RIFLE THE HYPOSPRAY GUARD ME FROM TEN MIUNTES AND KILL ALL OF SECURITY'

Doesn'T sound hard to me, latter is a multi-wish and therefore invalid. As well, kek 'let me steal the ion rifle' is literally a 'ok *disappears' wish. 

> 'm not even sure if you can make tazers stun something without knocking them over

No drop implant. Or whatever that one was that prevents you from falling over, ever.

In the end, all you bring up seems to be how people will abuse it (boink them, issue solved), how it ruins primary antags (the samw way Space Ninjas regularyly ruin blob and malf) and how it would make Djinn murderbone everything (which gets him banned). Not seeing the issue.

> Possible rather than rely on playerspeech, when you use it it opens up a menu panel where you can pick things

This would dismantle the whole concept of 'make a well-formulated wish'.

**Comment by ghost** [comment]
The fact that it's unkillable and you can't stop it from getting into antag hands means it's always gonna be a shuttle call with no way for crew to be triumphant.

**Comment by Alblaka** [comment]
> The fact that it's unkillable

Cept that killing it (which a laser gun or two can definitely achieve once it's phase is on CD) oneshots the Antag. In a very obvious fashion, too.

**Comment by ghost** [comment]
You can't kill it because it will keep respawning, as you said.

**Comment by Alblaka** [comment]
> You can't kill it because it will keep respawning, as you said.

Yes, but doing so
1. Invokes a cooldown of 5 minutes \* number of deaths. Meaning each success time removes the Djinn for longer.
2. After that it has to be found again. Which, in best case, will take less then a minute, in worst case, half a hour and several lamp respawns.
3. Kill (or well, crits plus sever item damage) whoever used it. Meaning you will usually remove an antag from the game (assuming you only kill the Djinn if it's actually being used to antag). Unless of course it was a ling, but lings are OP anyways.

If you want to really slap another nerf on the Djinn, declare it an always valid, regardless of circumstances.

**Comment by ghost** [comment]
How about just make it's death permanent?

Also, I'm still not understanding your point.

Would giving a wish to a djinn basically act like an AI law, where the person types in their wish? If so, I can see that working, I suppose, but it's still gonna be an issue.

Mostly because you're giving djinn free license to be a giant asshole, and trust me, people will take that and run with it. Unless we enforce a policy of 'no murderboning', you're going to see djinn interpret 'steal the ion rifle' as 'kill all of security and break every wall along the way and then steal the ion rifle and drag it into the singularity suicide bombing your master'

**Comment by Alblaka** [comment]
> Would giving a wish to a djinn basically act like an AI law, where the person types in their wish?

I imagined it more freeform like simply going 'say "Djinn, I order you to bring the nuclear authentication disk to this spot, without doing anything that would disclose my identity more then absolutely essentially, in the fastest time possible, without any other entity but you and me knowing where you bring it." ' 
Technically the djinn (if malicious) could still be an ass and bring you the disk in a locked briefcase, but due to the 'fastest time possible' constraint he cant take the 3 minutes to kill the clown and implant the disk into his chest to bring you a clown corpse. Nor can he rat you or tell where he is bringing the disk, and has to prevent anyone from following him, albeit he could reveal that he IS stealing the disk for his master (unnamed).

> Mostly because you're giving djinn free license to be a giant asshole, and trust me, people will take that and run with it. Unless we enforce a policy of 'no murderboning', you're going to see djinn interpret 'steal the ion rifle' as 'kill all of security and break every wall along the way and then steal the ion rifle and drag it into the singularity suicide bombing your master'

Yes, as I said, there will be rules for the respective Djinn. Example rules for the morally good and evil djinns respectively:

'Benevolent Djinn':
- Fullfill your master's wishes to your best interpretation of his intent.
- Do not hinder or harm your master in any way not explicitely stated as part of the wish.
- Do not harm, or cause permanent hindering, any other living being, unless doing so is absolutely necessary for your task.

With the evil being:
- Fullfill your master's wish as precisely as possible, yet against what you interpret to be his intent, as much as the formulation of the wish allows.
- Do not harm or hinder your master, unless doing so is an essential part of precisely following your master's wish.
- Do not go out of your task's why to harm or hinder other beings, but show no mercs to those trying to hinder or harm you.

And whilst we're att it, lets mash up some for the reverse djinn (who has a significantly lower chance of spawning then the other three)
- Always try to interpret your order in accordance with the masters's intent, then do your utmost to achieve the exact opposite as the intended result.
- Do not harm or hinder your master unless your master explicitely ordered you not to.
- Do not go out of your way to cause more chaos then inverting your master's intent actually requires.

(The spawning ratio of Djinn should be like {30/40/10/20}, making the 'malicious' Djinn (the one requiring careful formulation of your wish) the most common one, but making the inverse Djinn (who wiill plainly screw you over regardless of what you say) very rare. Note that the 'random djinn' (4th kind) rolls a 33/33/33 on which alignment he gets for each master seperately. In the end, if you make 'perfect' wishes every time, only the inverse djinn will be able to screw you over. And if you roll a beneficial djinn, even a simple order will usually work out)

**Comment by ghost** [comment]
If you make it 'say 'DJINN BRING ME EIGHT DONK POCKETS' rather than typing in the order yourself like a freeform law, there's no way for admins to tell what was said short of log diving. Djinn can say one thing, master can say it another, and without a way for djinn to look at the wish  they might forget and without a way for admins to look at the wish a nonantag master could wish for something murderboney, lie, and get the djinn banned.

**Comment by DavidEdwards** [comment]
It really shouldn't be common.

Something magical and game changing like this really needs to be heavily controlled. 

This is what I would suggest:
1. Not every round.
2. Once killed in a round - thats it for the Djinn. No respawning.
3. Wish must be given by an uneditable input field. Firstly, because admins need to easily see the wish. Secondly, in no film ever could someone change their mind about a given wish :D
4. Mischievious Djinnns need to be in the low probability area, otherwise people would probably just avoid it completely.
5. Djinn can be picked from the pool of ghosts / observers when rubbed.

**Comment by ghost** [comment]
Make it spawn an hour in, like ninja or blob.

Set chance of happening also to ninja or blob levels.

**Comment by Time-Green** [comment]
Maybe when you click the lamo again you get a popup, like making an AI law.
Then that will be the wish
